src/main/java/it/gov/pagopa/payhub/activities/connector/sendnotification/SendServiceImpl.java

```diff
@@ -36,4 +42,29 @@ public void deliveryNotification(String sendNotificationId) {
     public SendNotificationDTO notificationStatus(String sendNotificationId) {
         return sendClient.notificationStatus(sendNotificationId, authnService.getAccessToken());
     }
+
+    @Override
+    public SendNotificationDTO retrieveNotificationDate(String accessToken, String sendNotificationId, Long organizationId) {
```

Comment:
it's wrong that this logic is on the connector!
the connector should just perform the invocation of the send API!
The other logic should be moved on the Activity which invoked this API